public/scripts/project/lock.js

```diff
@@ -0,0 +1,75 @@
+define(function(require) {
+  var fs = Bramble.getFileSystem();
+  var sh = new fs.Shell();
+  var LOCK_KEY = 'bramble-lock';
+  var EXPIRY = 2000; // MS
+
+  function lock(root, callback) {
+    function refresh(callback) {
+      fs.setxattr(root, LOCK_KEY, Date.now(), function(err) {
+        if (err) {
+          return callback(err);
+        }
+
+        callback();
+      });
+    }
+
+    function setLock() {
+      function handleError(err) {
+        console.error("[Bramble] Failed to refresh project lock");
+      }
+
+      refresh(function(err) {
+        if (err) {
+          return callback(err);
+        }
+
+        setInterval(refresh.bind(null, handleError), EXPIRY);
```

Comment:
The thing that worries me about `setInterval` here is that the browser could bog down, and you could have 2 intervals overlap (i.e., a rush of intervals all at once).  I'd recommend a `setTimeout` that does another `setTimeout` when it runs, such that you don't have overlapping lock attempts.